langlib/lang.value/src/main/java/org/ballerinalang/langlib/value/ToJson.java

```diff
@@ -0,0 +1,344 @@
+        },
+        returnType = {
+                @ReturnType(type = TypeKind.JSON),
+                @ReturnType(type = TypeKind.ERROR)
+        },
+        isPublic = true
+)
+public class ToJson {
+
+    private static final String AMBIGUOUS_TARGET = "ambiguous target type";
+
+    public static Object toJson(Strand strand, Object value) {
+        if (value == null) {
+            return null;
+        }
+
+        BType sourceType = TypeChecker.getType(value);
+        BType targetJsonType = BTypes.typeJSON;
+        if (!isConvertibleToJson(value, new ArrayList<>())) {
+            if (sourceType.getTag() >= TypeTags.XML_ELEMENT_TAG || sourceType.getTag() == TypeTags.XML_TAG) {
+                return ToString.toString(strand, value);
+            } else {
+                throw createConversionError(value, targetJsonType);
+            }
+        }
+
+        // handle primitive values
+        if (sourceType.getTag() <= TypeTags.BOOLEAN_TAG) {
+            if (TypeChecker.checkIsType(value, targetJsonType)) {
+                return value;
+            } else {
+                // Has to be a numeric conversion.
+                return TypeConverter.convertValues(targetJsonType, value);
+            }
+        }
+
+        return convert((RefValue) value, targetJsonType, new ArrayList<>(), strand);
+    }
+
+    private static Object convert(RefValue value, BType targetType, List<TypeValuePair> unresolvedValues,
+                                  Strand strand) {
+        TypeValuePair typeValuePair = new TypeValuePair(value, targetType);
+
+        if (unresolvedValues.contains(typeValuePair)) {
+            throw new BallerinaException(CONSTRUCT_FROM_CYCLIC_VALUE_REFERENCE_ERROR,
+                    BLangExceptionHelper.getErrorMessage(RuntimeErrors.CYCLIC_VALUE_REFERENCE, value.getType()));
+        }
+
+        unresolvedValues.add(typeValuePair);
+
+        Object newValue;
+        switch (value.getType().getTag()) {
+            case TypeTags.MAP_TAG:
+            case TypeTags.RECORD_TYPE_TAG:
+                newValue = convertMapToJson((MapValue<?, ?>) value, targetType, unresolvedValues, strand);
+                break;
+            case TypeTags.ARRAY_TAG:
+            case TypeTags.TUPLE_TAG:
+                newValue = convertArrayToJson((ArrayValue) value, targetType, unresolvedValues, strand);
+                break;
+            case TypeTags.XML_TAG:
+            case TypeTags.XML_ELEMENT_TAG:
+            case TypeTags.XML_COMMENT_TAG:
+            case TypeTags.XML_PI_TAG:
+            case TypeTags.XML_TEXT_TAG:
+                newValue = ToString.toString(strand, value);
+                break;
+            case TypeTags.TABLE_TAG:
+                try {
+                    newValue = JSONUtils.toJSON((TableValueImpl) value);
+                } catch (Exception e) {
+                    return createConversionError(value, targetType, e.getMessage());
```

Comment:
This should be thrown right? Since the signature of `toJson()` is `json`

Reply:
Ack. Thanks